### PR TITLE
Enable SocketCAN interface tests in GitHub CI

### PR DESCRIPTION
**Pull request by lumagi**
See #1479
I still need to investigate why the CPython tests fail.

### CODE REVIEW

**Comment by lumagi** [comment]
For future reference: As of PR #1440 the default behavior for restarting Cyclic Send Tasks for SocketCAN was changed from raising an exception to not raising an exception. The test was still testing for an exception to be raised, which is no longer the case.

**Comment by lumagi** [comment]
@zariiii9003 I think this should be fine now. Please check if the CI integration and the tests suit you.

File: test/test_socketcan.py [hunk 1]
```diff
@@ -353,6 +355,21 @@ def test_build_bcm_update_header(self):
...
+        This test shall document raw CAN socket support under PyPy. Once this test fails, it is likely that PyPy
+        either implemented raw CAN socket support or at least changed the error that is thrown.
+        https://foss.heptapod.net/pypy/pypy/-/issues/3809
+        https://github.com/hardbyte/python-can/issues/1479
+        """
+        try:
+            can.Bus(interface="socketcan", channel="vcan0", bitrate=500000)
+        except OSError as e:
+            if "unknown address family" not in str(e):
+                raise
```
Comment (zariiii9003) [inline test/test_socketcan.py:371]: This test is a good idea, but let's change this to a warning message instead of a failing CI.

**Review by zariiii9003 — APPROVED** [review]
Really nice addition, thank you

**Pull request merged**